py4DSTEM/process/calibration/origin.py

```diff
@@ -224,6 +230,164 @@ def get_origin_from_braggpeaks(braggpeaks,Q_Nx,Q_Ny,findcenter='CoM',bvm=None):
+                                        boolean array. Mask will be applied to diffraction
+                                        patterns before finding the center.
+        probe_mask_std_scale (float):   size of Gaussian mask sigma. If set to None, function will estimate probe size.
+
+
+    Returns:
+        2 (R_Nx,R_Ny)-shaped ndarrays: the origin, (x,y) at each scan position
+    """
+
+    if probe_mask_size is None:
+        probe_mask_size, px, py = get_probe_size(np.fft.fftshift(probe_kernel))
+        probe_mask_size *= 2
+
+    # take fft of probe kernel
+    probe_kernel_FT = np.conj(np.fft.fft2(probe_kernel))
+
+    if qxyInit is None:
+        # find initial guess for probe center using mean diffraction pattern
+        diff_mean = np.mean(datacube.data,axis=(0,1))
+        peaks = _find_Bragg_disks_single_DP_FK(
+            diff_mean,
+            probe_kernel_FT,
+            corrPower = 1,
+            sigma = 0,
+            edgeBoundary = 0,
+            minRelativeIntensity = 0,
+            minAbsoluteIntensity = 0,
+            relativeToPeak = 0,
+            maxNumPeaks = 1,
+            subpixel = subpixel,
+            upsample_factor = upsample_factor)
+        qxyInit = np.array([peaks.data['qx'],peaks.data['qx']])
+
+    # Create mask
+    qx = np.arange(datacube.Q_Nx) - qxyInit[0]
+    qy = np.arange(datacube.Q_Ny) - qxyInit[1]
+    qya,qxa = np.meshgrid(qy,qx)
+    if mask is None:
+        mask = np.exp((qxa**2 + qya**2)/(-2*probe_mask_size**2))
+    else:
+        mask *= np.exp((qxa**2 + qya**2)/(-2*probe_mask_size**2))
+
+    # init output arrays
+    qx0_ar = np.zeros((datacube.R_Nx,datacube.R_Ny))
+    qy0_ar = np.zeros((datacube.R_Nx,datacube.R_Ny))
+
+    for (rx,ry) in tqdmnd(datacube.R_Nx,datacube.R_Ny,desc='Finding origins',unit='DP',unit_scale=True):
+    # for (rx,ry) in tqdmnd(2,2,desc='Finding origins',unit='DP',unit_scale=True):
+        if subpixel is None:
+            dp_corr = np.real(np.fft.ifft2(np.fft.fft2(datacube.data[rx,ry,:,:] * mask) * probe_kernel_FT))
+            ind2D = np.unravel_index(np.argmax(dp_corr), dp_corr.shape)
+            qx0_ar[rx,ry] = ind2D[0]
+            qy0_ar[rx,ry] = ind2D[1]
+        else:
+            peaks = _find_Bragg_disks_single_DP_FK(
+                datacube.data[rx,ry,:,:] * mask,
+                probe_kernel_FT,
+                corrPower = 1,
+                sigma = 0,
+                edgeBoundary = 0,
+                minRelativeIntensity = 0,
+                minAbsoluteIntensity = 0,
+                relativeToPeak = 0,
+                maxNumPeaks = 1,
+                subpixel = subpixel,
+                upsample_factor = upsample_factor)
+            qx0_ar[rx,ry] = peaks.data['qx']
+            qy0_ar[rx,ry] = peaks.data['qy']
+
+    return qx0_ar, qy0_ar
+
+def get_origin_autocorrelation(
+        datacube,
+        subpixel='multicorr',
+        upsample_factor=16,
+        mask=None):
```

Comment:
if this function doesn't work, i'd suggest removing it from the PR for now!

Reply:
i've removed this function - if it is functional at some future date, please PR again!

Reply:
As a note, I have added a previous version of this without subpixel to the ACOM branch, under the name `get_origin_single_dp_beamstop`